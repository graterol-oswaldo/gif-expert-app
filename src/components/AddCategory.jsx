import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };
  // senior: destructura target y solo referencia a partir de ahÃ­ a target
  //   const onInputChange = ({ target }) => {
  //     setInputValue(target.value);
  //   };

  const onSubmit = (event) => {
    event.preventDefault();
    // console.log(inputValue);
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form aria-label="form" onSubmit={(event) => onSubmit(event)}>
      <input
        type="text"
        placeholder="Search Gifs"
        value={inputValue}
        onChange={(event) => onInputChange(event)}
        // senior: envia event por default
        // onChange={onInputChange}
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
