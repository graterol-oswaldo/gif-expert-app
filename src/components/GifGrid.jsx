export const GifGrid = ({ category }) => {
  return (
    <div>
      <h3>{category}</h3>
      <p>Hola Mundo</p>
    </div>
  );
};
