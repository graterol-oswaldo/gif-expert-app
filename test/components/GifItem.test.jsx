import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

const title = 'saitama GIF';
const url = 'https://media0.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=3e5459b0og2rrkejg8y81tty3j2i6081jtabq03tneqbgmuc&rid=giphy.gif&ct=g'

describe('Testing <GifItem />', () => { 
    test('should match snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
     })

     test('should show the image with the URL and ALT corresponding', () => { 
        render(<GifItem title={title} url={url} />)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);
      })
      test('should show the paragraph corresponding', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText( title )).toBeTruthy();
        
       })
 })