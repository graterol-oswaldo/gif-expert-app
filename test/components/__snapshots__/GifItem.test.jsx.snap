// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama GIF"
      src="https://media0.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=3e5459b0og2rrkejg8y81tty3j2i6081jtabq03tneqbgmuc&rid=giphy.gif&ct=g"
    />
    <p>
      saitama GIF
    </p>
  </div>
</div>
`;
